src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -256,12 +272,37 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
             addr = self._inventory_get_fqdn(dd.hostname)
             peers.append(build_url(host=addr, port=port).lstrip('/'))
 
-        return {
-            "files": {
-                "alertmanager.yml": yml
-            },
-            "peers": peers
-        }, sorted(deps)
+        if self.mgr.secure_monitoring_stack and self.mgr.alertmanager_web_user and self.mgr.alertmanager_web_password:
+            deps.append(f'{hash(self.mgr.alertmanager_web_user + self.mgr.alertmanager_web_password)}')
```

Comment:
> for my info, why are we hashing this? Just to store a smaller bit of text?

No, we are hashing user/password to avoid exposing them as dependencies are logged in the cephadm debug logs.

Reply:
good point, forgot we logged the deps in certain situations